### PR TITLE
feat(route/dedao): add new dedao route 

### PR DESCRIPTION
**Pull request by Jacky-Chen-Pro**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/dedao/articles/8
/dedao/articles/9
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明
新增得到相关路由

### CODE REVIEW

**Comment by Jacky-Chen-Pro** [comment]
我想问一下ts中使用 any 这类检测错误，需要做修改和处理吗？是否影响合并

**Review by TonyRL — COMMENTED** [review]
Auto Route Test is failing

**Comment by Jacky-Chen-Pro** [comment]
> Auto Route Test is failing

Hi I have solve the problem, thank u

File: lib/routes/dedao/articles.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                    paragraph += content;
+                }
+            }
+            if (paragraph) {
+                paragraphs.push(`<p>${paragraph}</p>`);
+            }
+        }
+    }
+    return paragraphs.join('');
+}
```
Comment (TonyRL) [inline lib/routes/dedao/articles.ts:52]: You need a recursive function
```ts
function extractArticleContent(data) {
    if (!data || typeof data !== 'object') {
        return '';
    }

    let html = '';

    switch (data.type) {
        case 'paragraph':
            html += '<p>';
            if (data.contents && Array.isArray(data.contents)) {
                for (const item of data.contents) {
                    html += extractArticleContent(item);
                }
            }
            html += '</p>';
            break;

        case 'text':
            if (data.text) {
                let content = data.text.content || '';
                if (data.text.bold || data.text.highlight) {
                    content = `<strong>${content}</strong>`;
                }
                html += content;
            }
            break;

        default:
            break;
    }

    return html;
}
```
and call it in
```ts
JSON.parse(articleData.articleContent.content).map((data) => extractArticleContent(data)).join('')
```

Please render the `image` and `hr` as well.

**Review by TonyRL — COMMENTED** [review]
https://github.com/DIYgod/RSSHub/pull/17252/checks?check_run_id=32193205579

**Comment by Jacky-Chen-Pro** [comment]
> https://github.com/DIYgod/RSSHub/pull/17252/checks?check_run_id=32193205579

Thank you for your patient guidance and tips. I have modified the code, please review it when you have time.

File: lib/routes/dedao/articles.ts [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['igetget.com'],
+            target: '/articles/:id',
+        },
+    ],
+    name: '得到文章',
+    maintainers: ['qi12371'],
```
Comment (TonyRL) [inline lib/routes/dedao/articles.ts:26]: This user [does not exist](https://github.com/qi12371).
```suggestion
    maintainers: ['Jacky-Chen-Pro'],
```

Comment (Jacky-Chen-Pro) [reply lib/routes/dedao/articles.ts]: Sorry, this is my username on another platform. I have changed my GitHub name now.

Comment (Jacky-Chen-Pro) [reply lib/routes/dedao/articles.ts]: In fact, this is my first time submitting a pull request on GitHub. Thank you for your help!

**Pull request merged**